src/zhinst/toolkit/driver/devices/pqsc.py

```diff
@@ -150,23 +150,24 @@ def check_ref_clock(self, *, timeout: int = 30, sleep_time: int = 1) -> bool:
 
     def check_zsync_connection(
         self,
-        ports: Union[List[int], int] = 0,
+        ports: Union[List[int], int, List[BaseInstrument], BaseInstrument],
```

Comment:
I would rename the `ports` to something that reflects the variable better ... Maybe `input` ?